ownCloud/Messages/AlertView.swift

```diff
@@ -0,0 +1,217 @@
+		optionStackView.setContentHuggingPriority(.required, for: .vertical)
+		optionStackView.setContentHuggingPriority(.required, for: .horizontal)
+		optionStackView.distribution = .fillEqually
+		optionStackView.axis = .horizontal
+		optionStackView.spacing = 10
+
+		self.setContentCompressionResistancePriority(.required, for: .vertical)
+		self.setContentHuggingPriority(.required, for: .vertical)
+
+		self.addSubview(titleLabel)
+		self.addSubview(descriptionLabel)
+		self.addSubview(optionStackView)
+
+		headerLabel.setContentHuggingPriority(.required, for: .vertical)
+		titleLabel.setContentHuggingPriority(.required, for: .vertical)
+		descriptionLabel.setContentHuggingPriority(.required, for: .vertical)
+
+		headerLabel.setContentCompressionResistancePriority(.required, for: .vertical)
+		titleLabel.setContentCompressionResistancePriority(.required, for: .vertical)
+		descriptionLabel.setContentCompressionResistancePriority(.required, for: .vertical)
+
+		let enclosure = self.safeAreaLayoutGuide
+
+		if localizedHeader != nil {
+			headerContainer.translatesAutoresizingMaskIntoConstraints = false
+			headerContainer.setContentCompressionResistancePriority(.required, for: .vertical)
+			headerContainer.setContentHuggingPriority(.required, for: .vertical)
+
+			headerContainer.addSubview(headerLabel)
+			self.addSubview(headerContainer)
+
+			NSLayoutConstraint.activate([
+				headerLabel.leftAnchor.constraint(equalTo: headerContainer.leftAnchor, constant: 20),
+				headerLabel.rightAnchor.constraint(equalTo: headerContainer.rightAnchor, constant: -20),
+				headerLabel.topAnchor.constraint(equalTo: headerContainer.topAnchor, constant: 7),
+				headerLabel.bottomAnchor.constraint(equalTo: headerContainer.bottomAnchor, constant: -7),
+
+				headerContainer.topAnchor.constraint(equalTo: enclosure.topAnchor),
+				headerContainer.leftAnchor.constraint(equalTo: enclosure.leftAnchor),
+				headerContainer.rightAnchor.constraint(equalTo: enclosure.rightAnchor)
+			])
+		}
+
+		NSLayoutConstraint.activate([
+			titleLabel.topAnchor.constraint(equalTo: ((localizedHeader != nil) ? headerContainer.bottomAnchor : enclosure.topAnchor), constant: 20),
+			titleLabel.bottomAnchor.constraint(equalTo: descriptionLabel.topAnchor, constant: -5),
+			descriptionLabel.bottomAnchor.constraint(equalTo: optionStackView.topAnchor, constant: -20),
+			optionStackView.bottomAnchor.constraint(equalTo: enclosure.bottomAnchor, constant: -20),
+
+			titleLabel.leftAnchor.constraint(equalTo: enclosure.leftAnchor, constant: 20),
+			titleLabel.rightAnchor.constraint(equalTo: enclosure.rightAnchor, constant: -20),
+
+			descriptionLabel.leftAnchor.constraint(equalTo: enclosure.leftAnchor, constant: 20),
+			descriptionLabel.rightAnchor.constraint(equalTo: enclosure.rightAnchor, constant: -20),
+
+			optionStackView.leftAnchor.constraint(equalTo: enclosure.leftAnchor, constant: 20),
+			optionStackView.rightAnchor.constraint(equalTo: enclosure.rightAnchor, constant: -20)
```

Comment:
Would be better to use constants for all these values and font sizes!?

Reply:
Good point! Changed that.